app_backup.py

```
@@ -0,0 +1,415 @@
+
+    # Predict probability
+    try:
+        prob = model.predict_proba(full_input_df)[0][1]
+    except Exception as e:
+        st.error(f"Prediction error: {e}")
+        return None
+
+    # Use original (non-transformed) values for logic decisions
+    income = row.get('Monthly_Income_KES', 0)
+    repayment_history = None
+    has_collateral = row.get('Has_Collateral', None)
```

Ruff (0.12.2)

Use `row.get('Has_Collateral')` instead of `row.get('Has_Collateral', None)`

Replace `row.get('Has_Collateral', None)` with `row.get('Has_Collateral')`

(SIM910)